src/crypto/backup.ts

```diff
@@ -282,6 +284,24 @@ export class BackupManager {
         return this.checkAndStart();
     }
 
+    /**
+     * Attempts to retrieve a session from a key backup, if enough time
+     * has elapsed since the last check for this session id.
+     */
+    public async queryKeyBackupRateLimited(
+        targetRoomId: string | undefined,
+        targetSessionId: string | undefined,
+    ): Promise<void> {
+        const now = new Date().getTime();
```

Comment:
I think we're also missing a check to see if key backup is even enabled/possible? This whole area of code is a bit difficult to follow, but I think a null check on `this.backupInfo` or `this.algorithm` will do the job.

Reply:
Good catch, thanks!